### ISSUE TITLE
Declarative schema: Omitting indexType throws exception

### ISSUE THREAD

**Issue by k4emic** [magento/magento2#21322]
### Preconditions (*)

1. Magento 2.3
2.

### Steps to reproduce (*)

1. Add any module with the below db_schema.xml contents
2. Run `php bin/magento setup:upgrade` multiple times

```xml
<?xml version="1.0"?>
<schema xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:noNamespaceSchemaLocation="urn:magento:framework:Setup/Declaration/Schema/etc/schema.xsd">
    <table name="inventory_source_item">
        <column xsi:type="timestamp" name="updated_at" on_update="true" nullable="false" default="CURRENT_TIMESTAMP"
                comment="Update Time"/>
        <index referenceId="INVENTORY_SOURCE_ITEM_UPDATED_AT">
            <column name="updated_at"/>
        </index>
    </table>
</schema>
```
### Expected result (*)

1. Index is created on first run
2. Subsequent runs results in no modifications

### Actual result (*)

1. Index is created on first run  
2. Subsequent runs will throw an exception: `PDOException: SQLSTATE[42000]: Syntax error or access violation: 1061 Duplicate key name 'INVENTORY_SOURCE_ITEM_UPDATED_AT'`

**Comment by magento-engcom-team** [comment]
Hi @k4emic. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@k4emic do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
Hi @k4emic hi thank you for you report,  please open issue here ->https://github.com/magento-engcom/msi

**Comment by k4emic** [comment]
Hi @engcom-backlog-nazar 
The MSI component is simply used as an example here, adding an index to any table/column will result in this bug.

**Comment by k4emic** [comment]
Digging a bit into the code, this happens because the two compared elements (generated index and existing index) uses two different casings for "btree" in their dataset: https://cl.ly/f1b0ca1d4e8f/Image%202019-02-19%20at%208.50.19%20AM.png

`\Magento\Framework\Setup\Declaration\Schema\Dto\Factories\Index::DEFAULT_INDEX_TYPE` is set to "BTREE". Changing it to lowercase resolves the problem.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-backlog-nazar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-98315` were created

**Issue Available**: @engcom-backlog-nazar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @k4emic. Thank you for your report. 
The issue has been fixed in magento/magento2#21328 by @milindsingh in 2.3-develop branch
Related commit(s):
 - [b179e544a9c05eb0d5347028d4095755d019e304](https://github.com/magento/magento2/commit/b179e544a9c05eb0d5347028d4095755d019e304)

The fix will be available with the upcoming 2.3.2 release.
